interface/requests/response.go

```diff
@@ -6,172 +6,187 @@ package requests
-	RespPsaErrorAlreadyExists        uint16 = 1139
-	RespPsaErrorDoesNotExist         uint16 = 1140
-	RespPsaErrorInsufficientMemory   uint16 = 1141
-	RespPsaErrorInsufficientStorage  uint16 = 1142
-	RespPsaErrorInssuficientData     uint16 = 1143
-	RespPsaErrorCommunicationFailure uint16 = 1145
-	RespPsaErrorStorageFailure       uint16 = 1146
-	RespPsaErrorHardwareFailure      uint16 = 1147
-	RespPsaErrorInsufficientEntropy  uint16 = 1148
-	RespPsaErrorInvalidSignature     uint16 = 1149
-	RespPsaErrorInvalidPadding       uint16 = 1150
-	RespPsaErrorTamperingDetected    uint16 = 1151
+	StatusSuccess                      StatusCode = 0
+	StatusWrongProviderID              StatusCode = 1
+	StatusContentTypeNotSupported      StatusCode = 2
+	StatusAcceptTypeNotSupported       StatusCode = 3
+	StatusVersionTooBig                StatusCode = 4
+	StatusProviderNotRegistered        StatusCode = 5
+	StatusProviderDoesNotExist         StatusCode = 6
+	StatusDeserializingBodyFailed      StatusCode = 7
+	StatusSerializingBodyFailed        StatusCode = 8
+	StatusOpcodeDoesNotExist           StatusCode = 9
+	StatusStatusonseTooLarge           StatusCode = 10
+	StatusUnsupportedOperation         StatusCode = 11
+	StatusAuthenticationError          StatusCode = 12
+	StatusAuthenticatorDoesNotExist    StatusCode = 13
+	StatusAuthenticatorNotRegistered   StatusCode = 14
+	StatusKeyDoesNotExist              StatusCode = 15
+	StatusKeyAlreadyExists             StatusCode = 16
+	StatusPsaErrorGenericError         StatusCode = 1132
+	StatusPsaErrorNotPermitted         StatusCode = 1133
+	StatusPsaErrorNotSupported         StatusCode = 1134
+	StatusPsaErrorInvalidArgument      StatusCode = 1135
+	StatusPsaErrorInvalidHandle        StatusCode = 1136
+	StatusPsaErrorBadState             StatusCode = 1137
+	StatusPsaErrorBufferTooSmall       StatusCode = 1138
+	StatusPsaErrorAlreadyExists        StatusCode = 1139
+	StatusPsaErrorDoesNotExist         StatusCode = 1140
+	StatusPsaErrorInsufficientMemory   StatusCode = 1141
+	StatusPsaErrorInsufficientStorage  StatusCode = 1142
+	StatusPsaErrorInssuficientData     StatusCode = 1143
+	StatusPsaErrorCommunicationFailure StatusCode = 1145
+	StatusPsaErrorStorageFailure       StatusCode = 1146
+	StatusPsaErrorHardwareFailure      StatusCode = 1147
+	StatusPsaErrorInsufficientEntropy  StatusCode = 1148
+	StatusPsaErrorInvalidSignature     StatusCode = 1149
+	StatusPsaErrorInvalidPadding       StatusCode = 1150
+	StatusPsaErrorTamperingDetected    StatusCode = 1151
 )
 
-// ResponseBody represents a response body
+func (code StatusCode) IsValid() bool {
+	return (code >= StatusCode(0) && code <= StatusKeyAlreadyExists) || (code >= StatusPsaErrorGenericError && code <= StatusPsaErrorTamperingDetected)
+}
+
+// StatusonseBody represents a Statusonse body
 type ResponseBody struct {
 	*bytes.Buffer
 }
 
 // Response represents a Parsec response
 type Response struct {
-	Header WireHeader
-	Body   ResponseBody
+	Header *wireHeader
```

Comment:
Why the `*` here?

Reply:
being a pointer makes the assignment on line 89 simpler as the parse method will return nil value for wireHeader if it fails - no other reason than slightly cleaner code and avoiding a copy (i think)

Reply:
Ah I see. Well I was also going to propose to maybe change `parseWireHeaderFromBuf`
 to return a `wireHeader` instead? Then here we can have a `wireHeader` too to match the structure of the `Request`:
  ```go
type Request struct {
	Header wireHeader
	Body   RequestBody
	Auth   auth.RequestAuthToken
}
```
Or it's definitely more efficient to keep pointers everywhere?

Also, it is possible to keep the `Body` field? A `Response` is really a header + a body. If you think that is not needed here, I propose that we remove the `Response` struct altogether but I believe it would be nice to group them though.

Reply:
I don't want to sound too directive!! If you think it's more idiomatic to have pointers, let's keep them there!!

The only thing important here I think is to group the body (into the appropriate type that you think) and the header into a `Response`

Reply:
i've removed the response and responsebody types and refactored the NewResponse into ParseResponse - none of this was used downstream and wasn't used.  No point in having code with no real purpose (and don't think it future proofs anything)